crates/payload/basic/src/lib.rs

```diff
@@ -352,7 +377,15 @@ where
                     this.metrics.inc_failed_payload_builds();
                 }
                 Poll::Pending => {
-                    this.pending_block = Some(fut);
+                    #[cfg(not(feature = "optimism"))]
+                    {
+                        this.pending_block = Some(fut);
+                    }
+
+                    #[cfg(feature = "optimism")]
+                    if this.config.compute_pending_block {
+                        this.pending_block = Some(fut);
```

Comment:
I don't think `pending_block` should be `None` if `compute_pending_block` is false, see: https://github.com/ethereum-optimism/op-geth/blob/8f8af46ec728a79137d449891fc1827e1763cd7e/miner/worker.go#L401-L406

Reply:
The idea in Go is to represent it as nil everywhere, but map it to the fallback answer closer to the usage. Since the different usages may differ. The latest block is a fine fallback in the RPC, but internally you may want to know if it's just not available.